onmt/Models.py

```diff
@@ -113,55 +134,72 @@ def forward(self, src_input):
                                  .expand_as(emb))
             emb = self.dropout(emb)
 
-        out_length, out_batch, emb_size = emb.size()
+        out_length, out_batch, emb_dim = emb.size()
         aeq(in_length, out_length)
         aeq(in_length, out_length)
-        aeq(emb_size, self.embedding_size)
+        aeq(emb_dim, self.embedding_dim)
 
         return emb
 
+    @staticmethod
+    def create_instance(opt, dicts, for_encoder, feature_dicts=None):
+        """
+        Create an Embeddings instance.
+        """
+        if for_encoder:
+            embedding_dim = opt.src_word_vec_size
+        else:
+            embedding_dim = opt.tgt_word_vec_size
+        return Embeddings(embedding_dim,
+                          opt.position_encoding,
+                          opt.feat_merge,
+                          opt.feat_vec_exponent,
+                          opt.feat_vec_size,
+                          opt.dropout,
+                          dicts, feature_dicts)
+
 
 class Encoder(nn.Module):
     """
     Encoder recurrent neural network.
     """
-    def __init__(self, opt, dicts, feature_dicts=None):
+    def __init__(self, encoder_type, brnn, rnn_type,
+                 num_layers, rnn_size, dropout, enc_emb):
         """
         Args:
-            opt: Model options.
-            dicts (`Dict`): The src dictionary
-            features_dicts (`[Dict]`): List of src feature dictionaries.
+            encoder_type (string): rnn, brnn, mean, or transformer.
+            brnn (bool): brnn Encoder. Prefer `encoder_type=brnn` now.
```

Comment:
change this to `bidirectional` to match pytorch.  drop the "prefer" part of the comment